### PR TITLE
Release v0.6 revert tf1

### PR DESCRIPTION
**Pull request by vincentpierre**
Reverting the inference code to use TF#
 - Removed the Importer
 - Included the Tensor Definition
 - Included the TF# inference Engine
 - Converted the `.tf` models to `.bytes`
 - Reintroduced the `ENABLE_TENSORFLOW` flag
 - APIs were left unchanged

__TODO__
 - [x] Documentation __PR pending #1453__
 - [x] Testing

### CODE REVIEW

File: UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/TFSharpInferenceEngine.cs [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+                [DllImport("libtensorflow")]
+                private static extern unsafe void TF_OperationGetAttrType(IntPtr oper, string attr_name, 
+                        TFDataType* value, IntPtr status);
+                
+                [DllImport("libtensorflow")]
+                private static extern unsafe void TF_OperationGetAttrShape(IntPtr oper, string attr_name, long[] value, 
+                        int num_dims, IntPtr status);
+
+                [DllImport("libtensorflow")]
+                private static extern unsafe void TF_OperationGetAttrShapeList(
```
Comment (vladimiroster) [inline UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/TFSharpInferenceEngine.cs]: We're not using this method, so we can delete this code.

File: UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/TFSharpInferenceEngine.cs [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+                                                throw new FormatException("Could not query model for op shape (" + op.Name + ")");
+                                        }
+                                        else
+                                        {
+                                                shape = new long[dims.Length];
+                                                for (int i = 0; i < shape_attr.TotalSize; ++i)
+                                                {
+                                                        if (dims[i] == -1)
+                                                        {
+                                                                // TODO: since CoreML doesn't support batching yet
```
Comment (vladimiroster) [inline UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/TFSharpInferenceEngine.cs]: We can probably remove this comment

File: UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/Utils/RandomNormal.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
+﻿using System;
+
+namespace MLAgents.InferenceBrain.Utils
```
Comment (vladimiroster) [inline UnitySDK/Assets/ML-Agents/Scripts/InferenceBrain/Utils/RandomNormal.cs]: There are also unit tests for the utils classes, might worth porting them over as well.

**Comment by vincentpierre** [comment]
__Note__ I did force push because I added a commit about documentation by mistake to this PR. The documentation changes are on a different PR : #1453

File: UnitySDK/Assets/ML-Agents/Scripts/LearningBrain.cs [hunk 4]
```diff
@@ -135,9 +136,9 @@ protected override void DecideAction()
 #else
             if (agentInfos.Count > 0)
             {
-                throw new UnityAgentsException(string.Format(
-                    @"The brain {0} was set to inference mode but the Tensorflow 
-                        library is not present in the Unity project.",
+                Debug.LogError(string.Format(
```
Comment (vladimiroster) [inline UnitySDK/Assets/ML-Agents/Scripts/LearningBrain.cs]: Why this change?

Comment (vincentpierre) [reply UnitySDK/Assets/ML-Agents/Scripts/LearningBrain.cs]: If an error is thrown, the rest of the method is not called which causes a cascade of errors that made this exception hard to find. The quickest fix I could find was to log the error instead.

**Comment by vincentpierre** [comment]
Merge is dependent on #1453 being approved first

**Comment by xiaomaogy** [comment]
@vincentpierre Sorry for adding this comment so late. I know the feature is designed to be like this, but this might confuse our users. Currently when you set all of the agent to have PlayerBrain, and in the Broadcast Hub, you set the LearningBrain to Control flag checked, you can still do the fake training by using the "mlagents-learn ...", without any reward result. Maybe people would think they are doing the training, but actually it is in player mode. Should we output some kind of message to tell people that they are in this "broadcast mode"?
